### PR TITLE
C++ headers Integrity check - follows #10065

### PR DESCRIPTION
**Pull request by SSoulaimane**

### CODE REVIEW

**Review by jacob-carlborg — APPROVED** [review]

File: src/tests/cxxfrontend.c [hunk 1]
```diff
@@ -227,6 +227,14 @@ void test_semantic()
     semantic2(m, NULL);
     semantic3(m, NULL);
 
+    Loc loc;
+    Dsymbol *s = m->search(loc, Identifier::idPool("Error"));
```
Comment (ibuclaw) [inline src/tests/cxxfrontend.c]: You should be able to use `Loc()` here.

File: src/tests/cxxfrontend.c [hunk 2]
```diff
@@ -227,6 +227,14 @@ void test_semantic()
...
+    assert(s);
+    AggregateDeclaration *ad = s->isAggregateDeclaration();
+    assert(ad->ctor);
```
Comment (ibuclaw) [inline src/tests/cxxfrontend.c]: ad && ad->ctor

File: src/tests/cxxfrontend.c [hunk 3]
```diff
@@ -227,6 +227,14 @@ void test_semantic()
...
+    CtorDeclaration *ctor = ad->ctor->isCtorDeclaration();
+    assert(0 == strcmp(ctor->type->toChars(), "Error(string)"));
```
Comment (ibuclaw) [inline src/tests/cxxfrontend.c]: Maybe also:
assert(ctor->isMember() && !ctor->is Nested());

To test a couple more virtuals at random intervals in the vtable.

**Review by ibuclaw — CHANGES_REQUESTED** [review]

**Comment by SSoulaimane** [comment]
I also added two missing virtual function declarations to make the added tests succeed.

**Review by ibuclaw — APPROVED** [review]
OK

**Comment by ibuclaw** [comment]
> I also added two missing virtual function declarations to make the added tests succeed.

Good to see it caught more problems. :-)

**Pull request merged**